Asterix/tests/test_pupil.py

```diff
@@ -1,12 +1,17 @@
 from Asterix.optics import roundpupil
+import numpy as np
 
 
 def test_roundpupil():
     dim = 10
-    rad = int(dim / 2)
-    pupil = roundpupil(dim, rad, grey_pup_bin_factor=10)
+    rad = 4.3
 
-    zeros_left = (pupil[0, 0] == 0) and (pupil[0, 1] == 0) and (pupil[0, 2] == 0)
-    center = (pupil[0, 3] == 1) and (pupil[0, 4] == 1) and (pupil[0, 5] == 1) and (pupil[0, 6] == 1)
-    zeros_right = (pupil[0, 7] == 0) and (pupil[0, 8] == 0) and (pupil[0, 9] == 0)
-    assert zeros_left and center and zeros_right, f"{dim}-pixel pupil is not centered between pixels, in center of array."
+    pupil = roundpupil(dim, rad, grey_pup_bin_factor=2, center_pos='b')
+    centerb = (pupil == np.transpose(pupil)).all() and (pupil == np.flip(pupil, axis=0)).all()
+
+    assert centerb, f"{dim}-pixel pupil with center_pos='b' is not centered."
+
+    pupil = roundpupil(dim + 1, rad, grey_pup_bin_factor=3, center_pos='p')
+    centerp = (pupil == np.transpose(pupil)).all() and (pupil == np.flip(pupil, axis=0)).all()
+
+    assert centerp, f"{dim}-pixel pupil with center_pos='p' is not centered."
```

Comment:
Keep it simple. How about this:
```python
def test_roundpupil():
    dim = 10
    rad = 4.3

    pupil = roundpupil(dim, rad, grey_pup_bin_factor=2, center_pos='b')
    assert (pupil == np.transpose(pupil)).all(), f"{dim}-pixel pupil with center_pos='b' is not centered."
    assert (pupil == np.flip(pupil, axis=0)).all(), f"{dim}-pixel pupil with center_pos='b' is not centered."
    assert (pupil == np.flip(pupil, axis=1)).all(), f"{dim}-pixel pupil with center_pos='b' is not centered."

    pupil = roundpupil(dim + 1, rad, grey_pup_bin_factor=3, center_pos='p')
    assert (pupil == np.transpose(pupil)).all(), f"{dim}-pixel pupil with center_pos='p' is not centered."
    assert (pupil == np.flip(pupil, axis=0)).all(), f"{dim}-pixel pupil with center_pos='p' is not centered."
    assert (pupil == np.flip(pupil, axis=1)).all(), f"{dim}-pixel pupil with center_pos='p' is not centered."
```

Reply:
When a test fails, we need to know where exactly it fails, hence the separate assertions. And you can't really test too much so I am for testing both flip directions, as you added them.

Reply:
done. I removed the last one because it is redundant : transpose tests that top and left (and right and bottom) are equal and flip tests that left and right are equal. The second flip tests that the top and bottom is already ensured by the first 2